### PR TITLE
Purchases: Add a "reconnect" notice to disconnected sites

### LINKED ISSUE #12923: Manage Purchases: show a notice for disconnected sites with orphaned upgrades
With Jetpack's connection to WordPress.com it is possible for a user to purchase an upgrade, then disconnect the site from WordPress.com. This makes that purchase somewhat 'orphaned'. @ryelle has made it possible to display the disconnected site in the purchases list with the orphan purchase, but we can do the user one better.

Let's show a notice beneath the site/purchase block like this:

image

The reconnect link could jump them into step 2 of the Jetpack Connect flow (see if they have Jetpack installed or not and give them the appropriate instructions). 

cc @richardmuscat @beaulebens @johnHackworth 

Related to: #12746

### PR DESCRIPTION
**Pull request by ryelle**
This fixes #12923, giving users a way of reconnecting sites that have active subscriptions, but are disconnected from wp.com.

To test:

- Disconnect a jetpack site with a paid plan
- Visit `/me/purchases`
- You should see a banner under the purchase, asking you to reconnect:

screen shot 2017-04-24 at 4 44 56 pm

### CODE REVIEW

File: client/me/purchases/purchases-site/reconnect-notice.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	render() {
+		const { translate, name, domain } = this.props;
+
+		return (
+			<Notice
+				showDismiss={ false }
+				status="is-error"
+				text={ translate( '%(site)s has been disconnected from WordPress.com.', {
```
Comment (a8ci18n) [inline client/me/purchases/purchases-site/reconnect-notice.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( '%s has been disconnected.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ryelle** [comment]
Anyone testing might notice there's an issue with trying to click into disconnected site purchases — I've already reported that here: #13421

**Comment by sirreal** [comment]
How should this behave with multisite?

I have a multisite and my experience was the following:
* Main and subsite connected
* Buy plan on main site
* Disconnect main site
* ❓ **No reconnect banner for plan**
* Disconnect subsite
* ✅ **Banner displays for plan**

**Comment by sirreal** [comment]
I am not seeing the banner after I disconnect a site if purchases page has already shown the site.

1. Start with plan and site connected
1. Clear all application storage except cookies
1. Open /me/purchases
1. Navigate to site via UI
1. Disconnect via Calypso
1. Navigate to purchases via UI
1. ❌  Banner does not show, but plan exists and site is not connected. I have to clear storage and refresh to see the warning.

**Comment by ryelle** [comment]
Yeah, it seems we're not clearing disconnected sites from the redux list, I'm looking into this now.

… unfortunately, this also highlighted another problem — we only have the domain for disconnected sites, not the path, so the Reconnect button is linking to the wrong URL for subfolder installs (it links to connect `example.com`, not `example.com/wp`).

**Comment by rickybanister** [comment]
Visually this is 💯

**Comment by ryelle** [comment]
I think the refreshing data issue is tied to the sites-list removal, and as that wraps up, it'll clear this up too — if you test this PR in conjunction with #13094, it works as expected, disconnecting a site immediately shows it as "disconnected" in the purchase list.

File: client/me/purchases/purchases-site/reconnect-notice.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	render() {
+		const { translate, name, isJetpack } = this.props;
+		let text = translate( 'You are no longer a user on %(site)s and cannot manage this purchase.', {
+			args: {
+				site: name,
+			}
+		} );
+		if ( isJetpack ) {
+			text = translate( '%(site)s has been disconnected from WordPress.com.', {
```
Comment (a8ci18n) [inline client/me/purchases/purchases-site/reconnect-notice.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( '%s has been disconnected.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ryelle** [comment]
It looks like the "Reconnect" button is out of scope for now (we only have the domain in a purchase object, even at the API level, so on subfolder installs the button would be connecting to the wrong URL). I've tried instead the following, where the action is now "Contact Support", linking off to Jetpack support for JP sites, and wp.com support for wp.com sites (as the same thing should happen on wp.com sites where you've made a purchase, and since been removed from the site)

screen shot 2017-05-05 at 5 50 19 pm

I also removed the link into disconnected purchases, which can be restored once #13421 is fixed.

**Comment by sirreal** [comment]
@ryelle Would you like me to test again, or are we waiting on #13094 still?

**Comment by ryelle** [comment]
@sirreal Can you test again? The disconnected site not refreshing isn't related to this PR - I'm pretty sure it's happening on production too. I was hoping the other PR would be merged faster, but it's understandably taking time, so I'd like to get this one in sooner :)

**Comment by ryelle** [comment]
@rickybanister can you look this over again for a design ✅ ? This https://github.com/Automattic/wp-calypso/pull/13354#issuecomment-299584515 should tell you what to check.

**Comment by rickybanister** [comment]
This looks great to me. Merge it.

**Pull request merged**